### PR TITLE
[INCOMPATIBLE] docs(man): add initial version of ctags-optlib(7)

### PR DESCRIPTION
**Pull request by masatake**
@k-takata, can I ask you to review this.

### CODE REVIEW

File: man/ctags-optlib.7.rst.in [hunk 1]
```diff
@@ -18,4 +18,289 @@ SYNOPSIS
...
+FLAGS FOR ``--regex-<LANG>`` OPTION
+-----------------------------------
+
+You can specify more than one flag at the end of ``--regex-<LANG>`` to
+control how Universal-ctags uses the pattern.
+
+Exuberant-ctags uses a *letter* to represent a flag. In
+Universal-ctags, a *name* surrounded by braces can be used instead of
+the *letter*. The name form makes reading an optlib file easier. The
+most of ail flags newly instead in Universal-ctags don't have the
```
Comment (k-takata) [inline man/ctags-optlib.7.rst.in]: `ail`: Mistake of `all`?
`instead`: should be `added` or something???

File: man/ctags-optlib.7.rst.in [hunk 2]
```diff
@@ -18,4 +18,289 @@ SYNOPSIS
...
+	``--kind-<LANG>`` option. If ``K`` field is enabled, the *name*
+	is used as value in the "kind" extension field in tags output.
+
+	The *description* and *letter* are listed in ``--list-kinds``
+	output. All three elements of kind-spec are listed in
+	``--list-kinds-full`` output. Don't use braces in the
+	*description*. They will be used meta characters in the future.
+
+``--regex-<LANG>=/regexp/replacement/[kind-spec/][flags]``
+	The /regexp/replacement/ pair define a regular expression
```
Comment (k-takata) [inline man/ctags-optlib.7.rst.in]: `define` -> `defines`
(maybe)

File: man/ctags-optlib.7.rst.in [hunk 3]
```diff
@@ -18,4 +18,289 @@ SYNOPSIS
 DESCRIPTION
 -----------
 
-TO BE WRITTEN.
+**Exuberant-ctags**, the ancestor of **Universal-ctags**, has provided
+the way to define a new parser from command line.  Universal-ctags
+extends, refines this feature. **optlib parser** is the name for such
```
Comment (k-takata) [inline man/ctags-optlib.7.rst.in]: `extends,` -> `extends and`
(maybe)

File: man/ctags-optlib.7.rst.in [hunk 4]
```diff
@@ -18,4 +18,289 @@ SYNOPSIS
...
+   concepts(definition, reference, kind, role, field, extra). About
+   the concepts, ctags(1) of Universal-ctags may help you.
+
+2. Give a file pattern or file extension for activating the parser
+
+   Use ``--map-<LANG>=[+|-]extension|pattern``.
+
+3. Give a name to the parser
+
+   Use ``--langdef=name`` option. *NAME* is refereed as *<LANG>* in
```
Comment (k-takata) [inline man/ctags-optlib.7.rst.in]: `refereed` -> `referred`

File: man/ctags-optlib.7.rst.in [hunk 5]
```diff
@@ -18,4 +18,289 @@ SYNOPSIS
...
+   the later steps.
+
+4. Define kinds
+
+   Use ``--kinddef-<LANG>=letter,name,description`` option.
+   Universal-ctags introduces this option.  Exuberant-ctags doesn't
+   have. In Exuberant-ctags, a kind is defined as a side effect of
+   specifying ``--regex-<LANG>=`` option. So user doesn't have a
+   change to recognize how important the definition of kind.
```
Comment (k-takata) [inline man/ctags-optlib.7.rst.in]: `change`: Maybe a mistake of `chance`?

File: man/ctags-optlib.7.rst.in (hunk 2 above)
Comment (masatake) [reply man/ctags-optlib.7.rst.in]: You are correct. I found many "... pair IS" in my corpuses.

**Comment by masatake** [comment]
@k-takata, thank you very much. I updated.

**Comment by masatake** [comment]
I added more changes about ``--regex-<LANG>=...,file,``. Some of them introduces incompatibility.

**Comment by masatake** [comment]
Squashed.

**Pull request merged**